### ISSUE TITLE
Identify if focus is console or text editor

### ISSUE THREAD

**Issue by rjake** [posit-dev/positron#3941]
I want to know if I'm currently in the console or in a text editor. I used to use `getActiveDocumentContext()` to determine this and either send code to the console or add the code to the file I am editing. Right now, irunning this function from the console and text editor return the same list ofvalues. I see that `$id` has been identified as an issue (below), but not sure about the other items (`$path`, etc). 

I saw these issues and PRs that seem related:
* https://github.com/posit-dev/ark/pull/237
* https://github.com/posit-dev/positron/issues/2191
* https://github.com/posit-dev/positron/issues/2571 - focuses on `$id` 
* https://github.com/posit-dev/positron/pull/2582

I'm not sure how it all works under the hood but the [VS Code context keys](https://code.visualstudio.com/api/references/when-clause-contexts#available-context-keys) `editorFocus` and `terminalFocus` might be able to help.

Here's a comparison between **Positron** _(great name btw)_ and **RStudio**
* ![image](https://github.com/posit-dev/positron/assets/14003618/faada6e4-b9e0-450d-9a1c-b823c59907cd)

**Comment by juliasilge** [comment]
Yep, this is mostly the same as #2571, and we also will need to look at `path` at that time as well. We can leave this issue open for the examples to check. Thanks!

Side note: we do have the context key `positronConsoleFocused` which tells you if, as it says, the console is focused. You can use this today in keyboard shortcuts, tasks, etc like `"when": "positronConsoleFocused"` but don't currently expose this via the rstudioapi R package functions.

**Comment by klmr** [comment]
Hi, apologies for dragging this up. Is there a timeline to fix this? I’m getting requests from multiple users to implement Positron support, and unfortunately the current behaviour is blocking this (klmr/box#374).

(I also feel that the issue label is misleading: the current behaviour of `rs.api.getActiveDocumentContext()` is a bug, not an enhancement; especially since it’s inconsistent with the behaviour in RStudio.)

**Comment by juliasilge** [comment]
No timeline planned currently, but input from users is really helpful for prioritization.

What specifically that gets returned there is not working for your use case? Are you needing the `id` there that gets returned? Or is it the `path` that is a problem?

**Comment by klmr** [comment]
@juliasilge Yes, the issue is the path. In a nutshell, ‘box’ needs to know what “path” it’s invoked from to locate files stored relative to it.

`box::use(./foo)` is expected to load code from a file `foo.r` that is stored in the same directory as the currently-executing code. So, if the expression is executed from a file (by whatever means: `source()`, “Run” button, keyboard shortcut in editor, …) it needs to be resolved relative to that file. ‘box’ is using `.rs.api.getActiveDocumentContext()` to find that path.

But if a user manually enters the expression in the console, the user expectation is different: the code should behave independently of the file that currently happens to be shown (in particular, the meaning of the code should absolutely not change after the user has switched the view to a different file), and should instead resolve relative to the current directory (or, potentially, the project root; although this isn’t how ‘box’ currently handles it).

RStudio behaves in exactly the right way here: when executing `.rs.api.getActiveDocumentContext()` inside console, it returns an empty context (since the code wasn’t executed from the context of a document).
